gradle.properties

```diff
@@ -1,7 +1,17 @@
-org.gradle.jvmargs = -Xmx2G
-kotlin.code.style  = official
-
 # Project properties
-projectName        = Weave-Loader
-projectGroup       = net.weavemc
-projectVersion     = 1.0.0-b.2
\ No newline at end of file
+projectName = Weave-Loader
+projectGroup = net.weavemc
+projectVersion = 1.0.0-b.2
+
+# Gradle properties
+org.gradle.daemon = true
+org.gradle.caching = true
+org.gradle.parallel = true
+org.gradle.configureondemand=true
+org.gradle.parallel.threads=4
+org.gradle.dependency.verification = strict
+org.gradle.kotlin.dsl.allWarningsAsErrors = true
+org.gradle.configuration-cache = true
```

Comment:
Personally not a huge fan of the configuration cache, since, in my experience, taking a different path through the task graph results in a full reconfigure (even on buildSrc level) when this is not required when disabling the configuration cache: not sure if that was just a weird issue on my end, but we should test and reconsider using this feature.

Reply:
> taking a different path

What do you mean? This should just cache previously discovered paths. I observed ridiculous increases in compilation speeds with it, but I can definitely if you found any issues with it

Reply:
A project I worked on a while ago used configuration caching, and it would reconfigure the entire project if I ran `clean build` instead of solely `build`. I dunno, maybe just a skill issue on my end.

Reply:
`clean build` seems to work :+1: